### ISSUE TITLE
MaxPooling2d error?

### ISSUE THREAD

**Issue by marcohekla** [fchollet/keras#4145]
Hi,

I have an issue with the MaxPooling2d layers in Keras using theano (and in theano itself).

I fail to run my code when I use a MaxPooling2d layer in Keras (using theano). I have my own neural network code using theano which also fails when I use the theano maxpooling:

from theano.tensor.signal import pool
pool_layer = pool.pool_2d()

Perhaps there is an issue in this new maxpooling version in theano? I believe that it was recently changed, previously being:
from theano.tensor.signal import downsample

Basically running the MNIST convolutional example fails with the maxpooling layer in Keras, but works without. You can see the simple code here:
https://github.com/fchollet/keras/blob/master/examples/mnist_cnn.py

I would very much appreciate any help. Since everything else is working for me, I have an idea that this is a general issue.

Thanks,
Marco

**Comment by nouiz** [comment]
What is the error message? Without that it is hard to help.

Make sure to use Theano dev version.

**Comment by marcohekla** [comment]
Hi, the error is the following:

![maxpool_error](https://cloud.githubusercontent.com/assets/23002157/19737472/ea86a7c8-9bb3-11e6-89e3-480f218b8aa2.png)

Everything is working fine in my conv network, except the maxpooling. Any suggestions would be great - thanks.

**Comment by nouiz** [comment]
This is a partial error message. Can you redirect the output of the python
script to a file and send this? The current error don't have enough
information.

**Comment by marcohekla** [comment]
Hi Nouiz, 

Thanks for taking the time to help me out!
 
One question before I attempt getting the full error message; can you make maxpooling work? An easy way to test it would be to use the following code example on the MNIST data set:
https://github.com/fchollet/keras/blob/master/examples/mnist_cnn.py

If you (or anyone else in here?) can use the maxpooling, then there is something wrong with my setup..

I just tried to redirect the error message using the following code:

-----------------------------------------------------
import sys
sys.stderr = open('errorlog.txt', 'w')

#do stuff

sys.stderr.close()
sys.stderr = sys.__stderr__

-----------------------------------------------------

It gave me the errorlog.txt attached here.

Best regards,
Marco
[errorlog.txt](https://github.com/fchollet/keras/files/556506/errorlog.txt)
